### ISSUE TITLE
Malformed JSON/XML output

### ISSUE THREAD

**Issue by martinrlie** [vinumeris/lighthouse#157]
I've also hit the [previously reported](https://groups.google.com/d/topic/lighthouse-discuss/sgRRE48wP2A/discussion) problems with my Lighthouse server's JSON/XML output, and wanted to register it in this issue list.

Apparently, it's an [error in the profobuf library](https://code.google.com/p/protobuf-java-format/issues/detail?id=45), that Lighthouse uses.

**Comment by mikehearn** [comment]
Yes. Better to just use protobufs directly. It's a much simpler format. I'll look for a better protobuf->json converter lib.

**Comment by martinrlie** [comment]
Data-wise, I'm sure you're right, but most developers will use libraries to read even such small structures as this (i.e. it's the easy of including/using the library that matters, not the data format itself). They will have experience with JSON/XML and be reluctant to parse binary formats "manually" - or add a new library only for this. Basically: It adds a barrier to developing simple things on top of Lighthouse, like simple status pages, etc.

Could a temporary "workaround" be removing the "orig_hash" property/element if the requested format isn't binary?

Rationale: It's quite possibly a quick fix, you're [already doing some scrubbing](https://github.com/vinumeris/lighthouse/blob/master/server/src/main/java/lighthouse/server/ProjectHandler.java#L206-L230) while adding the pledges, "orig_hash" is known to cause problems which causes parsers to fail reading the entire output, and a working list of pledges without the hash is probably a lot better than a list which will "randomly" fail (with the hash).

**Comment by martinrlie** [comment]
(If you agree, I might even try send a PR myself. If I can get the project to build, that is - I haven't even tried that, so far. :D)

**Comment by mikehearn** [comment]
A patch would be welcome! The problem is there can be arbitrary binary data fields not just orig_hash. A better fix would be to change the json library to just encode binary fields as hex or base64 rather than trying to do escaping (or use/find a new one). 

That said, JSON is just an unbelievably poor data format and the more people we can get onto protobufs the better IMO. If it weren't for the limitations of web browsers I'd be tempted to just scrap support for it entirely.
